### PR TITLE
Fixup DeepSpeed things

### PR DESCRIPTION
**Pull request by muellerzr**
# What does this PR do?

Fixes a (soon to be failing) DS test with accelerate 1.0. We need to manually override the DS plugin when doing hp sweeps since we now don't allow for arbitrary passing in of a ds plugin to an already initialized accelerator

~~Also notes an actual deprecation point for when we want to force users to use `integrations.deepspeed` rather than `.deepspeed` to 5.0. @ArthurZucker @LysandreJik let me know if that's too soon, but it's something that was irking me :)~~

Seems we already got rid of it! 🔥 

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@SunMarc @LysandreJik

### CODE REVIEW

File: src/transformers/trainer.py [hunk 1]
```diff
@@ -1674,7 +1675,11 @@ def _hp_search_setup(self, trial: Union["optuna.Trial", Dict[str, Any]]):
 
             self.args.hf_deepspeed_config = HfTrainerDeepSpeedConfig(self.args.deepspeed)
             self.args.hf_deepspeed_config.trainer_config_process(self.args)
-            self.args.deepspeed_plugin = DeepSpeedPlugin(hf_ds_config=self.args.hf_deepspeed_config)
+
+            # From 1.0 on, we also need to fully reset/override the DS plugin when doing sweeps
+            if AcceleratorState._shared_state != {} and version.parse(accelerate_version) > version.parse("0.34.2"):
+                self.args.deepspeed_plugin = None
+                AcceleratorState().deepspeed_plugins = DeepSpeedPlugin(hf_ds_config=self.args.hf_deepspeed_config)
```
Comment (SunMarc) [inline src/transformers/trainer.py:1682]: I think that the else condition is missing

**Review by SunMarc — APPROVED** [review]
LGTM ! Left a comment

File: src/transformers/trainer.py (hunk 1 above)
Comment (muellerzr) [reply src/transformers/trainer.py:1682]: Prior to 1.0 it didn't matter at all, so the only new blocker is with 1.0.

Comment (muellerzr) [reply src/transformers/trainer.py:1682]: Post offline discussion, technically we should be able to do `AcceleratorState().reset_state(False)` here instead and it should do what we want without issue. Will try this alternative (then we don't need to limit it to a 1.0 flag either)

**Review by LysandreJik — APPROVED** [review]
SGTM!

**Pull request merged**